test/addStyles.test.js

```diff
@@ -0,0 +1,154 @@
+"use strict";
+var assert = require("assert");
+var url = require('url');
+
+describe("addStyles update transform function", function() {
+  var originalCSSLike = '.validCSS.required {}';
+  var updatedCSSLike = '.updatedCSS.required {}';
+  var emptyCSSLike = '';
+  var window;
+
+  var originalList = [['module_path', originalCSSLike, '']];
+  var updatedList = [['module_path', updatedCSSLike, '']];
+  var emptyList = [['module_path', emptyCSSLike, '']];
+
+  // a lot of internal mutation
+  var addStylesEntry = require("../lib/addStyles");
+
+  var utils = require("./utils"),
+              runDOMTest = utils.runDOMTest;
+
+  var jsdomHtml = [
+    "<html>",
+    "<head>",
+    "</head>",
+    "<body>",
+    "<div class='target'>",
+    "</div>",
+    "</body>",
+    "</html>"
+  ].join("\n");  
+
+  afterEach(function() {
+    window = undefined;
+  }); 
+
+  // With pain, mutate the global window object 
+  it("should verify basic transform capability during HMR", function(done) {
+    var oneShot = 1;
+    var styleLoaderOptions = {};
+    styleLoaderOptions.transform = function(css) {
+      return oneShot-- && css;
+    };
+
+    function applyUpdate() {
+
+      //Mutate global scope as the addStyles code expects it
+      global.window = window = this;
+      global.document = this.document;
+      // Unfortunately getElement internally memoizes at mutuates at the module level so we
+      // can't test with it. This gets us around the issue
+      styleLoaderOptions.insertInto = function() {
+        return window.document.head;
+      };
+
+      var updateFn = addStylesEntry.call(window, originalList, styleLoaderOptions);
+      // After the initial load, we should have a style node
+      assert(this.document.querySelectorAll('style').length);
+
+      updateFn.call(window, updatedList);
+      // After the update, the transform returns falsey we should remove it
+      const numStyleNodes = this.document.querySelectorAll('style').length;
+
+      // force it to clean up the stylesInDom
+      updateFn.call(window);
+
+      return numStyleNodes;
+    }
+
+    runDOMTest(
+      jsdomHtml,
+      undefined,
+      0,
+      done,
+      applyUpdate
+    )
+   });
+
+  // See issue #324, place holder for a future test once that issue is fixed
+  it.skip("should load the css if the transform function returns data during HMR");
```

Comment:
What actually we need to solve this problem?

Reply:
The discussion we had over on issue #324 there is quite a bit of refactoring. I can remove the test if you want but I am not in a position to spend more time on this at present

Reply:
@labriola you can left this test, it is normal, we should add more tests to increase coverage